### ISSUE TITLE
frictionless schema obscures NaN values

### ISSUE THREAD

**Issue by iskandari** [adokter/bioRad#608]
compare:

```bash
❯ grep 4400 temp/csv/bewid_vpts_20230414.csv | cut -d ',' -f 4-9 | head -n 6
NaN,NaN,NaN,NaN,NaN,NaN
NaN,NaN,NaN,NaN,NaN,NaN
NaN,NaN,NaN,NaN,NaN,NaN
NaN,NaN,NaN,NaN,NaN,NaN
NaN,NaN,NaN,NaN,NaN,NaN
NaN,NaN,NaN,NaN,NaN,NaN
```

```r
library(frictionless)

url="https://aloft.s3-eu-west-1.amazonaws.com/baltrad/daily/bewid/2023/bewid_vpts_20230414.csv"
file_name <- basename(url)
temp_file <- file.path(tempdir(), file_name)
curl::curl_download(url, temp_file)

package <- frictionless::create_package()
schema <- "https://raw.githubusercontent.com/enram/vpts-csv/main/vpts-csv-table-schema.json"
package <- frictionless::add_resource(package,"vpts",data = temp_file, schema = schema)
vpts_df <- frictionless::read_resource(package, "vpts")
head(vpts_df[vpts_df$height==4400,][4:9])
# A tibble: 6 × 6
#u     v     w    ff    dd sd_vvp
#<dbl> <dbl> <dbl> <dbl> <dbl>  <dbl>
#1    NA    NA    NA    NA    NA     NA
#2    NA    NA    NA    NA    NA     NA
#3    NA    NA    NA    NA    NA     NA
#4    NA    NA    NA    NA    NA     NA
#5    NA    NA    NA    NA    NA     NA
#6    NA    NA    NA    NA    NA     NA
```

@peterdesmet is it expected that the [vpts-csv-table-schema](https://github.com/enram/vpts-csv/blob/main/vpts-csv-table-schema.json) also contains NaN under missing values?  If so, how can one differentiate between empty values and NaN as they are both interpreted as NA?  This can be solved with:

```r
package$resources[[1]]$schema$missingValues <- c("", "NA") 
```
but wanted to raise the issue as it seems important to be able to differentiate nodata an undetect.

Related to #590

**Comment by peterdesmet** [comment]
@iskandari indeed, this is caused by the line in the vpts table schema you reference, where NaN is listed as a missing value.

The discussion that lead to this is here: https://github.com/enram/vpts-csv/issues/39#issuecomment-1410953452. I wrote there:

> 2. CSV readers would still read `NaN` into memory as `NaN`, since we don't refer to the table schema for reading data, only for validating data.

But that is no longer true, since we do use frictionless to read the data (which comes with a number of benefits, such as casting the data to the correct type, handling of multiple files and errors).

The solutions I see are:

1. Not listing `NaN` as missing value in the specs, which would require us to remove all min/max constraints to pass validation.
2. Reading the table schema first with, dropping `NaN` from the missing values as suggested above and then reading the data. This will retain the `NaN` values:

``` r
library(magrittr)
library(jsonlite)
library(frictionless)

url <- "https://aloft.s3-eu-west-1.amazonaws.com/baltrad/daily/bewid/2023/bewid_vpts_20230414.csv"
schema <- "https://raw.githubusercontent.com/enram/vpts-csv/main/vpts-csv-table-schema.json"
schema_updated <- jsonlite::fromJSON(schema, simplifyDataFrame = FALSE, simplifyVector = TRUE)
# Alternatively, use private function frictionless:::read_descriptor(schema)
schema_updated$missingValues <- c("", "NA")

p <-
  frictionless::create_package() %>%
  frictionless::add_resource("as_is", data = url, schema = schema) %>%
  frictionless::add_resource("updated", data = url, schema = schema_updated)

as_is <- frictionless::read_resource(p, "as_is")
head(as_is[as_is$height==4400,][4:9])
#> # A tibble: 6 × 6
#>       u     v     w    ff    dd sd_vvp
#>   <dbl> <dbl> <dbl> <dbl> <dbl>  <dbl>
#> 1    NA    NA    NA    NA    NA     NA
#> 2    NA    NA    NA    NA    NA     NA
#> 3    NA    NA    NA    NA    NA     NA
#> 4    NA    NA    NA    NA    NA     NA
#> 5    NA    NA    NA    NA    NA     NA
#> 6    NA    NA    NA    NA    NA     NA

updated <- frictionless::read_resource(p, "updated")
head(updated[updated$height==4400,][4:9])
#> # A tibble: 6 × 6
#>       u     v     w    ff    dd sd_vvp
#>   <dbl> <dbl> <dbl> <dbl> <dbl>  <dbl>
#> 1   NaN   NaN   NaN   NaN   NaN    NaN
#> 2   NaN   NaN   NaN   NaN   NaN    NaN
#> 3   NaN   NaN   NaN   NaN   NaN    NaN
#> 4   NaN   NaN   NaN   NaN   NaN    NaN
#> 5   NaN   NaN   NaN   NaN   NaN    NaN
#> 6   NaN   NaN   NaN   NaN   NaN    NaN
```

<sup>Created on 2023-06-26 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by iskandari** [comment]
To avoid refactoring with the first solution and maintain the original schema, we can go with the second, fixed in #590 941a4779b651e1b1a5d11e0d67f8fad7586ab3e6
